### PR TITLE
[SMTChecker] Support tuples as multi var decl

### LINKED ISSUE #4141: SMT: Check that tuple assignments are handled properly.

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes #4141 #5380
Depends on #6652 

This is the last PR in the tuple series

### CODE REVIEW

**Comment by chriseth** [comment]
Please rebase.

**Comment by leonardoalt** [comment]
Will rebase after the other one is merged.

**Comment by leonardoalt** [comment]
Rebased

File: libsolidity/formal/SMTChecker.cpp [hunk 1]
```diff
@@ -330,20 +330,35 @@ bool SMTChecker::visit(ForStatement const& _node)
...
 	if (_varDecl.declarations().size() != 1)
-		m_errorReporter.warning(
-			_varDecl.location(),
-			"Assertion checker does not yet support such variable declarations."
-		);
-	else if (knownVariable(*_varDecl.declarations()[0]))
+	{
+		if (auto init = _varDecl.initialValue())
+		{
+			auto const& symbTuple = dynamic_pointer_cast<SymbolicTupleVariable>(m_expressions[init]);
```
Comment (chriseth) [inline libsolidity/formal/SMTChecker.cpp]:
```suggestion
			auto symbTuple = dynamic_pointer_cast<SymbolicTupleVariable>(m_expressions[init]);
```
(I think the other version should also work, but better not confuse people - `dynamic_pointer_cast` returns a copy by value)

**Review by chriseth — DISMISSED** [review]

**Review by chriseth — APPROVED** [review]

**Pull request merged**